scrunch/scripts.py

```diff
@@ -44,19 +46,50 @@ def get_default_syntax_flag(self, strict_subvariable_syntax):
         """
         if strict_subvariable_syntax is not None:
             return strict_subvariable_syntax
-        flags = self.dataset_resource.session.feature_flags
+        flags = self.resource.session.feature_flags
         return flags.get("clients_strict_subvariable_syntax", DEFAULT_SUBVARIABLE_SYNTAX)
 
+    def execute(self, script_body, strict_subvariable_syntax=None):
+        pass
+
+
+class SystemScript(BaseScript):
+
+    def format_request_url(self, execute, strict_subvariable_syntax=None):
+        strict_subvariable_syntax_flag = self.get_default_syntax_flag(strict_subvariable_syntax)
+        if strict_subvariable_syntax_flag:
+            execute.self += "?strict_subvariable_syntax=true"
+
+    def execute(self, script_body, strict_subvariable_syntax=None):
+        """
+        Will run a system script on this.
+
+        System scripts do not have a return value. If they execute correctly
+        they'll finish silently. Otherwise, an error will raise.
+        """
+        # The project execution endpoint is a shoji:view
+        payload = shoji_view_wrapper(script_body)
+        try:
+            execute = self.resource.execute
+            self.format_request_url(execute, strict_subvariable_syntax)
+            return execute.post(payload)
```

Comment:
I see why you had to mutate the instance, I think that here it's best to use a raw request, see how this pycrunch `.post()` is implemented and you'll do a `self.resource.session.post(execute_url. ....)`.

Reply:
As explained above, I did not mutate the instance, so we should be safe